Source/Serializer.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  Serializer.swift
+//  KakapoExample
+//
+//  Created by Alex Manzella on 31/03/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+public protocol Serializable {
+    func serialize() -> Any
+}
+
+public extension Serializable {
+    public func serialize() -> Any {
+        return Kakapo.serialize(self)
+    }
+}
+
+extension Array: Serializable {
+    public func serialize() -> Any {
+        var array = [Any]()
+        for obj in self {
+            array.append(serializeObject(obj))
+        }
+        return array
+    }
+}
+
+extension Dictionary: Serializable {
+    public func serialize() -> Any {
+        var dictionary = [Key: Any]()
+        for (key, value) in self {
+            dictionary[key] = serializeObject(value)
+        }
+        return dictionary
+    }
+}
+
+extension Optional: Serializable {
+    public func serialize() -> Any {
+        switch self {
+        case let .Some(value):
+            return serializeObject(value)
+        default:
+            return NSNull()
+        }
+    }
+}
+
+extension _PropertyPolicy {
+    public func serialize() -> Any {
+        return serializeObject(_object)
+    }
+}
+
+private func serializeObject(value: Any) -> Any {
+    if let value = value as? Serializable {
+        return value.serialize()
+    } else {
+        assert(value is AnyObject) // TODO: throw
```

Comment:
Then, could you please add a test that tests this assertion? I guess a struct that does not conform to Serializable, since other value types are automatically bridged?